src/components/members/filters/CheckboxModal.js

```diff
@@ -0,0 +1,59 @@
+import styles from "./modals.module.scss";
+import {Box, Button, Control, Modal, ModalOverlay} from "@unioncredit/ui";
+import {useEffect, useState} from "react";
+import {ReactComponent as FilterIcon} from "images/filter.svg";
+
+export const CheckboxModal = ({id, open, title, values, filters, handleClose}) => {
+  const [selected, setSelected] = useState([]);
+
+  const handleApplyFilters = () => {
+    filters.addCheckboxFilter(id, selected);
+    handleClose();
+  }
+
+  const handleOnChecked = (value) => {
+    if (selected.includes(value)) {
+      setSelected(selected.filter(v => v !== value));
+    } else {
+      setSelected([...selected, value]);
+    }
+  }
+
+  useEffect(() => {
+    const item = filters.checkboxValues.find(v => v.key === id);
+    setSelected(item ? item.values : []);
+  }, [filters.checkboxValues]);
```

Comment:
Assuming this is an array this dep is going to cause this to re-run everytime because:
```js
[1, 2, 3] == [1, 2, 3] // false
```

Unless this filters that is being passed in is always memoized eg `useMemo`.

A dirty fix is just pass `[JSON.stringify(filters.checkboxValues)]`